### PR TITLE
Add OpenStreetMap example using TileLayer and BitmapLayer

### PR DESCRIPTION
**Pull request by xintongxia**
For #2802 

#### Change List
- example
openstreet

### CODE REVIEW

**Comment by Pessimistress** [comment]
Screenshot?

File: examples/website/openstreet-map/app.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            this.setState({
+              [layerId]: {
+                image: data.image,
+                bounds: data.bounds
+              }
+            });
+          });
+
+          return new BitmapLayer({id: layerId, ...this.state[layerId]});
+        }
```
Comment (xintongxia) [inline examples/website/openstreet-map/app.js]: From my understanding, `renderSubLayers` callback makes sense when the SubLayer expects `data` field, deck.gl Layer auto resolves the `data` promise and feed resolved data to that SubLayer. 

But `BitmapLayer` does not rely on data, and it uses `image` and `bounds` to place the image. So I have to manipulate the react state to trigger the layer re-render when data resolved.

File: examples/website/openstreet-map/app.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import React, {Component} from 'react';
+import {render} from 'react-dom';
+
+import DeckGL, {TileLayer, BitmapLayer} from 'deck.gl';
+
+export const INITIAL_VIEW_STATE = {
+  latitude: 47.65,
+  longitude: 7,
+  zoom: 4.5,
+  maxZoom: 16,
```
Comment (Pessimistress) [inline examples/website/openstreet-map/app.js]: Any reason to set a max zoom? Comment?

File: examples/website/openstreet-map/app.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  _renderLayers() {
+    return [
+      new TileLayer({
+        renderSubLayers: props => {
+          const {x, y, z} = props.tile;
+
+          return new BitmapLayer({
+            id: `bitmap-${props.id}`,
+            image: `https://${s}.tile.openstreetmap.org/${z}/${x}/${y}.png`,
+            bounds: tileIndexToBounds({x, y, z})
```
Comment (Pessimistress) [inline examples/website/openstreet-map/app.js]: Just a thought: the lnglat bounds of a tile could be something we provide as part of the `Tile` class. I can see a lot of use cases needing these numbers.

File: examples/website/openstreet-map/index.html [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  <style>
+    body {
+      margin: 0;
+      font-family: sans-serif;
+      width: 100vw;
+      height: 100vh;
+      overflow: hidden;
+    }
+
+    .tooltip {
```
Comment (Pessimistress) [inline examples/website/openstreet-map/index.html]: Idea: auto highlight the tile under the cursor and show a tooltip of its x,y,z coordinates.

File: examples/website/openstreet-map/package.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  "license": "MIT",
+  "scripts": {
+    "start-local": "webpack-dev-server --env.local --progress --hot --open",
+    "start": "webpack-dev-server --progress --hot --open"
+  },
+  "dependencies": {
+    "deck.gl": "^7.0.0-alpha",
+    "react": "^16.3.0",
+    "react-dom": "^16.3.0",
+    "react-map-gl": "^3.3.0"
```
Comment (Pessimistress) [inline examples/website/openstreet-map/package.json]: This can be removed now?

File: modules/layers/src/bitmap-layer/bitmap-layer.js [hunk 6]
```diff
@@ -192,7 +198,12 @@ export default class BitmapLayer extends Layer {
 
     if (typeof image === 'string') {
       loadImage(image).then(data => {
-        this.setState({bitmapTexture: new Texture2D(gl, {data})});
+        this.setState({
+          bitmapTexture: new Texture2D(gl, {
+            data,
+            parameters: DEFAULT_TEXTURE_PARAMETERS
```
Comment (Pessimistress) [inline modules/layers/src/bitmap-layer/bitmap-layer.js]: Does this help the image quality?

File: examples/website/openstreet-map/app.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+export const INITIAL_VIEW_STATE = {
+  latitude: 47.65,
+  longitude: 7,
+  zoom: 4.5,
+  maxZoom: 16,
+  pitch: 50,
+  bearing: 0
+};
+
+const s = 'c';
```
Comment (Pessimistress) [inline examples/website/openstreet-map/app.js]: What does this mean?

File: examples/website/openstreet-map/app.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+export class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
+  _renderLayers() {
+    return [
+      new TileLayer({
```
Comment (Pessimistress) [inline examples/website/openstreet-map/app.js]: What are the min and max zoom of OSM tiles? Let's set them explicitly.

File: modules/layers/src/bitmap-layer/bitmap-layer.js (hunk 6 above)
Comment (xintongxia) [reply modules/layers/src/bitmap-layer/bitmap-layer.js]: Not really. It looks the same with or without these parameters.

**Comment by TrantorM** [comment]
I tried to run your openstreet-map example and got an error shown in the attached image. To use the example standalone I deleted the bottom line `module.exports = env...` in the file `webpack.config.js` according the description within that file.

Bildschirmfoto 2019-03-28 um 14 59 08

Any Idea?

**Comment by ibgreen** [comment]
@TrantorM FWIW:

If you are testing things on master, you probably want to use `npm run start-local` to build the example against the local source on master rather than against installed modules.

The reason is that since deck.gl v7 is still in development the published modules may not contain what you need.

**Comment by TrantorM** [comment]
Not able to build it on master with `npm run start-local` either. Can't resolve '@loaders.gl/core'.

Bildschirmfoto 2019-03-29 um 12 12 13

A `git checkout tags/v7.0.0-beta.1`  results in the same problem.

**Pull request merged**